src/middlewares/auth.js

```diff
@@ -1,6 +1,6 @@
 const auth = async (req, res, next) => {
     if (!req.session.user) {
-        return res.redirect('/login')
+        return res.status(401).redirect('/login')
```

Comment:
While this approach is fine , it better to use a controller to handle authentication. 
Also using jwt is the best approach. 
[](https://github.com/auth0/node-jsonwebtoken)

create a controller.js or create a controller folder and handle all calls 
verify user , generate a nonce for that user and then create jwt

Reply:
example `.then((accessToken: string) => res.json({ accessToken }))
			.catch(next)`